### ISSUE TITLE
Having trouble with ActiveRecord::ConnectionTimeoutError in a clean ruby app

### ISSUE THREAD

**Issue by kalopilato** [klarna/phobos_db_checkpoint#29]
Hi there,

I've run up a new ruby app with your `phobos` gem, happily consuming kafka events from a local kafka server.  

After adding the `phobos_db_checkpoint` gem (following the readme, adding a custom handler including `PhobosDBCheckpoint::Handler`) I'm repeatedly seeing this error:
```
ERROR -- Phobos : <Hash> {:message=>"error processing message, waiting 0.42s",
:waiting_time=>0.42, :exception_class=>"ActiveRecord::ConnectionTimeoutError", 
:exception_message=>"could not obtain a connection from the pool within 5.000 seconds 
(waited 5.003 seconds); all pooled connections were in use", ... }
``` 
originating from `PhobosDBCheckpoint::Handler#around_consume`.

I was able to manually persist events using `PhobosDBCheckpoint::Event.new` when debugging.

I've closely followed the docs for both gems and spent several hours troubleshooting - do you have any ideas as to what may be causing this?  My code is here:
https://github.com/kalopilato/kafka_event_notifier

Thanks!

**Comment by tulios** [comment]
Hi @kalopilato, we can take a look. Apparently your configuration is correct, we checked our own projects and we are configuring the pool in the `database.yml`, but the `pool_size` option in the `configure` method should have the same effect. I will try to run your project and see what I find.

Thanks for the report!

**Comment by kalopilato** [comment]
Great thanks @tulios!  The `configure(pool_size)`call was just a quick sanity check to try a connection pool size bigger than the default; we'll definitely configure it in `database.yml` once we get things working 👍

**Comment by kalopilato** [comment]
Hey @tulios, we did a little more debugging this morning and found that `PhobosDBCheckpoint.load_db_config()` is getting called three times on startup - once from our `phobos_boot.rb`, and two more times from `tasks.rb` - so our `pool_size` is getting overriden.  

We found that patching `phobos_db_checkpoint.rb:47` from
```
pool_size = Phobos.config.listeners.map { |listener| listener.max_concurrency || 1 }.inject(&:+)
```
to
```
pool_size = Phobos.config.listeners.map { |listener| listener.max_concurrency || 1 }.inject(&:+) + 1
```
seemed to resolve the issue. (We noticed that there was a connection sitting idle, using up our pool with default size of 1).

In the meantime we have a workaround, calling `PhobosDBCheckpoint.configure()` a second time from `phobos_boot.rb`, setting the pool size again as per this commit: https://github.com/kalopilato/kafka_event_notifier/commit/a2f3dbad982ab82fb6420beb618e51ff523f62ee
Hope that helps!

**Comment by tulios** [comment]
@kalopilato sorry for the delay, I was super busy last week. Your solution works but I would like to fix `load_db_config` to keep the configured `pool_size`. If you define the pool size in the `database.yml` it will also work. I'm glad that you are unblocked. I'll try to provide a more definitive solution this week.

I've created a new issue to track this bug: #30

Thanks for all the debugging!

**Comment by klippx** [comment]
This should be fixed in v3.2.0 @kalopilato can you please verify and/or close the ticket if you agree?
